Libs/MRML/Core/vtkMRMLColorNode.cxx

```diff
@@ -375,6 +375,23 @@ const char *vtkMRMLColorNode::GetColorName(int ind)
 }
 
 //---------------------------------------------------------------------------
+int vtkMRMLColorNode::GetColorIndexByName(const char *name)
+{
+  if (this->GetNamesInitialised() && name != NULL)
+    {
+    std::string strName = name;
+    for (unsigned int i = 0; i < this->Names.size(); i++)
```

Comment:
To be consistent, either use public APIs (this->GetNumberOfColors() and this->GetColorName()) or use member variables directly (this->Names.size() and this->Names[i]).

Also, a minor thing: get number of colors only once and store it in a variable.